### ISSUE TITLE
ESM packaging exports WithCSSVar type

### ISSUE THREAD

**Issue by gluck** [chakra-ui/chakra-ui#4085]
# 🐛 Bug report

Since 1.11.0, esm (JS) build of `@chakra-ui/styled-system` wrongly re-exports the type `WithCSSVar`:

In `https://unpkg.com/browse/@chakra-ui/styled-system@1.11.0/dist/esm/index.js`:
```
export { WithCSSVar } from "./utils/types";
```

(version 1.10.5 didn't have this export)

`./utils/types` having no exported variables (types were pruned at build time), this breaks latest rollup version from packaging this version of the library, rolling back to 1.10.5 being the workaround.
 
## 💥 Steps to reproduce

1. Go to https://stackblitz.com/edit/chakra-rollup-repro
2. Run `npm run build` in terminal
3. See error:
```
[!] Error: 'WithCSSVar' is not exported by node_modules/@chakra-ui/styled-system/dist/esm/utils/types.js, imported by node_modules/@chakra-ui/styled-system/dist/esm/index.js
```

(optionaly)
4. Fork, revert to version 1.10.5 / run / no error

## 💻 Link to reproduction

https://stackblitz.com/edit/chakra-rollup-repro

## 🧐 Expected behavior

Having no re-exports of pruned variables

## 🧭 Possible Solution

Not familiar with the build chain used here, but a workaround could be to use `export *` which will work even with types pruned.

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Chakra UI / styled-system        |    1.11.0        |
| Browser          |      Chrome / Not relevant      |
| Operating System |      Linux / Not relevant      |

**Comment by cschroeter** [comment]
Thanks for reporting the issue.

I'm happy to say that this has already been fixed in fd231f7

**Comment by gluck** [comment]
Great news, thanks.
Sorry for the extra report, I could've searched in the PRs as well.
